api/auth.py

```diff
@@ -0,0 +1,41 @@
+"""JWT authenticator decorator for endpoints."""
+import json
+import jwt
+from user_actions import user_exists
+
+
+def jwt_auth(handler_function):
+    """Authenticates JWT against User DB"""
+
+    def _require_auth(self, *args, **kwargs):
+        token = self.request.headers.get_list('Authorization').pop()
+        if token:
+            parts = token.split()
+            if parts[0].lower() != 'bearer':
+                throw_authorization_error(self)
+            elif len(parts) == 1:
+                throw_authorization_error(self)
+            elif len(parts) > 2:
+                throw_authorization_error(self)
+
+            jwt_token = parts[1]
+            decoded_token = jwt.decode(jwt_token, self.settings['jwt_secret'])
+            if user_exists(decoded_token['email']):
+                handler_function(self, *args, **kwargs)
+            else:
+                throw_authorization_error(self)
+        else:
+            throw_authorization_error(self)
```

Comment:
I don't think it's 100% necessary, but it might be nice to be able to customize these authorization messages a bit. Every one of these calls to `throw_authorization_error` is going to respond with the same message. Being able to pass in the message as an argument could be helpful?